src/V3SchedVirtIface.cpp

```diff
@@ -141,124 +109,59 @@ class VirtIfaceVisitor final : public VNVisitor {
-        }
+        iterateChildren(nodep);
     }
     void visit(AstLoopTest* nodep) override {
         unsupportedWriteToVirtIface(nodep->condp(), "loop condition");
     }
-    void visit(AstJumpBlock* nodep) override {
-        {
-            VL_RESTORER(m_trigAssignp);
-            VL_RESTORER(m_trigAssignIfacep);
-            VL_RESTORER(m_trigAssignMemberVarp);
-            iterateChildren(nodep);
-        }
-        if (v3Global.usesTiming()) {
-            // Clear the trigger assignment, as there could have been timing controls in the jump
-            // block
-            m_trigAssignp = nullptr;
-            m_trigAssignIfacep = nullptr;
-            m_trigAssignMemberVarp = nullptr;
-        }
-    }
     void visit(AstNodeStmt* nodep) override {
-        if (v3Global.usesTiming()
-            && nodep->exists([](AstNode* nodep) { return nodep->isTimingControl(); })) {
-            m_trigAssignp = nullptr;
-            m_trigAssignIfacep = nullptr;
-            m_trigAssignMemberVarp = nullptr;
-        }
-        FileLine* const flp = nodep->fileline();
-
-        foreachWrittenVirtIfaceMember(nodep, [&](AstVarRef*, AstIface* ifacep,
-                                                 AstVar* memberVarp) {
-            if (ifacep != m_trigAssignIfacep || memberVarp != m_trigAssignMemberVarp) {
-                // Write to different interface member than before - need new trigger assignment
-                m_trigAssignIfacep = ifacep;
-                m_trigAssignMemberVarp = memberVarp;
-                m_trigAssignp = nullptr;
+        VL_RESTORER(m_curStmt);
+        m_curStmt = nodep;
+        iterateChildren(nodep);
+    }
+    void visit(AstVarRef* const nodep) override {
+        if (!m_curStmt) return;
+        if (nodep->access().isReadOnly()) return;
+        AstIface* ifacep = nullptr;
+        AstVar* memberVarp = nullptr;
+        if (AstIfaceRefDType* const dtypep = VN_CAST(nodep->varp()->dtypep(), IfaceRefDType)) {
+            if (dtypep->isVirtual()) {
+                if (AstMemberSel* const memberSelp = VN_CAST(nodep->firstAbovep(), MemberSel)) {
+                    // Extract the member varp from the MemberSel node
+                    memberVarp = memberSelp->varp();
+                    ifacep = dtypep->ifacep();
+                }
             }
-            if (!m_trigAssignp) {
-                m_trigAssignp
-                    = new AstAssign{flp, createVirtIfaceMemberTriggerRefp(flp, ifacep, memberVarp),
-                                    new AstConst{flp, AstConst::BitTrue{}}};
-                nodep->addNextHere(m_trigAssignp);
+        } else if ((ifacep = nodep->varp()->sensIfacep())) {
+            memberVarp = nodep->varp();
+        } else if (VN_IS(nodep->backp(), AssignW)) {
+            memberVarp = nodep->varScopep()->varp();
+            AstNode* backp = memberVarp;
+            while ((backp = backp->backp())) {
+                if ((ifacep = VN_CAST(backp, Iface))) break;
```

Comment:
Can you instead use m_modp, or otherwise earlier remember the interface when it was visited in the hierarchy?